src/main.rs

```diff
@@ -1,45 +1,105 @@
-
-#[tokio::main]
-async fn main() -> Result<(), Error> {
-    let token = env::var("TELEGRAM_BOT_TOKEN").expect("TELEGRAM_BOT_TOKEN not set");
-    let bot_name = env::var("TELEGRAM_BOT_USERNAME").expect("TELEGRAM_BOT_USERNAME not set");
-    let re = Regex::new(r"(http://www\.|https://www\.|http://|https://)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(/.*)?").unwrap();
-    let vs = vec!["Вилкой в глаз или в жопу раз?", "Когда на небе была очередь за умом или сиськами, ты стоял в очереди за пирожком", "Впервые вижу членовагину", "Do you like сосать?"];
-    let api = Api::new(token);
-
-    // Fetch new updates via long poll method
-    let mut stream = api.stream();
-    while let Some(update) = stream.next().await {
-        // If the received update contains a new message...
-        let update = update?;
-        println!("{:?}", update);
-        if let UpdateKind::Message(message) = update.kind {
-            if let MessageKind::Text { ref data, .. } = message.kind {
-                // Print received text message to stdout.
-                println!("<{}>: {}", &message.from.first_name, data);
-                if_chain! {
-	                if let Some(ref reply_to_message) = message.reply_to_message;
-	                if let MessageOrChannelPost::Message(mess) = &**reply_to_message;
-	                if mess.from.username.as_ref().unwrap_or(&"".to_owned()) == &bot_name;
-	                then {
-			                api.send(message.text_reply(vs[rand::thread_rng().gen_range(0..vs.len())]))
-			                .await?;
-	        	 	} else {
-		                if re.is_match(data) {
-			                api.send(message.text_reply(vs[rand::thread_rng().gen_range(0..vs.len())]))
-			                .await?;
-		                }
-	        	 	}
-	            }
-            }
+use clap::{App, SubCommand};
+use crate::toxic_bot::ToxicBot;
+use std::{fs, env};
+use std::fs::{File};
+use crate::constants::*;
+use std::error::Error;
+use std::io::{BufReader, BufRead};
+use whatlang::Lang;
+use crate::telegram_bot::TelegramBot;
+
+mod toxic_bot;
+mod cmd_bot;
+mod constants;
+mod telegram_bot;
+
+fn main() {
+    env_logger::init();
+
+    let matches = App::new("ToxicBot")
+        .about("A bot that will roast you")
+        .author("Bohdan Iakymets <cyberbond95@gmail.com>, Serhii Zakharov <serhii@zahar.pro>")
+        .subcommand(
+            SubCommand::with_name("cmd")
+                .about("just for testing, or if you're feeling \
+                        like you want to be roasted and don't have Telegram")
+        )
+        .subcommand(
+            SubCommand::with_name("telegram")
+                .about("to run telegram bot. \
+                        You'll also need to set TELEGRAM_BOT_TOKEN env variable")
+        )
+        .get_matches();
+
+    let mut bot = ToxicBot::new();
+
+    load_dir(&mut bot, DEFAULT_DATA_PATH).unwrap();
+
+    match matches.subcommand_name() {
+        Some("cmd") => cmd_bot::run(&mut bot),
```

Comment:
Nit-pick: for consistency sake, let's use camel case as you have with `TelegramBot``

Reply:
Well, cmd_bot is a module and we can't use camel case according to the style guide, we could also create a struct here, but I don't want to overengineere it even more :D